### PR TITLE
Add Storefront endpoint for payment instrument's setup

### PR DESCRIPTION
**Pull request by 2e3s**
Setting up a payment instrument with a setup transaction through a new endpoint `payment-instruments/{id}/setup` (GET and POST a related transaction).
It looks somewhat RPCish, but in essence it's still resource-based. This approach is more decoupled from the payment instrument.
Alternative to #618

### CODE REVIEW

File: openapi/components/requestBodies/storefront/SetupPaymentInstrumentRequest.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        - websiteId
+        - currency
+      properties:
+        websiteId:
+          description: The website identifier string.
+          allOf:
+            - $ref: ../../schemas/ResourceId.yaml
+        currency:
+          allOf:
+            - $ref: ../../schemas/CurrencyCode.yaml
```
Comment (adamaltman) [inline openapi/components/requestBodies/storefront/SetupPaymentInstrumentRequest.yaml:15]:
```suggestion
          $ref: ../../schemas/CurrencyCode.yaml
```

**Comment by 2e3s** [comment]
devs please review

**Comment by 2e3s** [comment]
devs please review

**Comment by 2e3s** [comment]
devs please review

**Review by VladimirBerdnik — APPROVED** [review]

**Comment by VladimirBerdnik** [comment]
devs please review

**Review by ashkarpetin — APPROVED** [review]

**Pull request merged**